### ISSUE TITLE
[BUG] Crashing after updating to latest debug build

### ISSUE THREAD

**Issue by D3SOX** [Deletescape-Media/Lawnchair#4175]
### Describe the bug

App crashing on start

### Steps to reproduce

Upgrade to latest dev build as of now

### Expected behavior

Don't crash

### Device information

- Device: Google Pixel 8 Pro
- OS: CalyxOS Android 14
- App version: fd5058a91ff18d581f12a068bc64d665cecfb39c

### Additional context

```
Lawnchair (Debug) bug report Apr 2, 2024 8:51:53 PM
version: 14 Dev (#668) (14)
commit: fd5058a
build.brand: google
build.device: husky
build.display: AP1A.240305.019.A1.24505020
build.fingerprint: google/husky/husky:14/AP1A.240305.019.A1/11445699:user/release-keys
build.hardware: husky
build.id: AP1A.240305.019.A1
build.manufacturer: Google
build.model: Pixel 8 Pro
build.security.level: 2024-03-05
build.product: husky
build.type: user
version.codename: REL
version.incremental: 24505020
version.release: 14
version.sdk_int: 34
display.density_dpi: 360
isRecentsEnabled: false

error: Uncaught exception

java.lang.RuntimeException: Unable to start activity ComponentInfo{app.lawnchair.debug/app.lawnchair.LawnchairLauncher}: java.lang.ClassCastException: com.android.launcher3.widget.LauncherWidgetHolder$HolderFactory cannot be cast to app.lawnchair.factory.LawnchairWidgetHolder$LawnchairHolderFactory
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3946)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4126)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:108)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:195)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:157)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:90)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2544)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8501)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: java.lang.ClassCastException: com.android.launcher3.widget.LauncherWidgetHolder$HolderFactory cannot be cast to app.lawnchair.factory.LawnchairWidgetHolder$LawnchairHolderFactory
	at app.lawnchair.LawnchairLauncher.createAppWidgetHolder(LawnchairLauncher.kt:357)
	at com.android.launcher3.Launcher.onCreate(Launcher.java:544)
	at com.android.launcher3.uioverrides.QuickstepLauncher.onCreate(QuickstepLauncher.java:582)
	at app.lawnchair.LawnchairLauncher.onCreate(LawnchairLauncher.kt:223)
	at android.app.Activity.performCreate(Activity.java:8767)
	at android.app.Activity.performCreate(Activity.java:8745)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1519)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3928)
	... 13 more
```

**Comment by zer0fl0w** [comment]
same. pixel 7 and s20.

**Comment by mvmike** [comment]
Faced the same issue (pixel 7), downgrading to the previous `14-dev` build fixed it → https://github.com/LawnchairLauncher/lawnchair/actions/runs/8508603619

**Comment by D3SOX** [comment]
What's your method of downgrading? Are you rooted? I can't downgrade because I don't have CorePatch on my new phone (not rooted)

**Comment by mvmike** [comment]
Previous build has the same version code (technically you're not downgrading), there is no need to be rooted or anything.

**Comment by D3SOX** [comment]
Oh, never mind. The build from the link you provided worked. I tried a different one before and thought it generally doesn't work.

**Comment by Goooler** [comment]
Should be fixed by 286f5b5cc0e6e2695b497969e06a7bc167222d31.
